test/e2e/prometheus_test.go

```diff
@@ -1070,6 +1070,79 @@ func TestPromOpMatchPromAndServMonInDiffNSs(t *testing.T) {
 	}
 }
 
+func TestThanos(t *testing.T) {
+	t.Parallel()
+
+	ctx := framework.NewTestCtx(t)
+	defer ctx.Cleanup(t)
+	ns := ctx.CreateNamespace(t, framework.KubeClient)
+	ctx.SetupPrometheusRBAC(t, ns, framework.KubeClient)
+
+	peerServiceName := "thanos-peers"
+	querierServiceName := "thanos-querier"
+	basicPrometheus := framework.MakeBasicPrometheus(ns, "basic-prometheus", "test-group", 1)
+	basicPrometheus.Spec.Thanos = &monitoringv1.ThanosSpec{
+		Peers:   fmt.Sprintf("%s.%s.svc:10900", peerServiceName, ns),
+		Version: "v0.1.0-rc.0",
+	}
+	pservice := framework.MakePrometheusService(basicPrometheus.Name, "test-group", v1.ServiceTypeClusterIP)
+	tservice := framework.MakeThanosService(peerServiceName)
+	qservice := framework.MakeThanosQuerierService(querierServiceName)
+	s := framework.MakeBasicServiceMonitor("test-group")
+	thanosQuerier, err := testFramework.MakeDeployment("../../example/thanos/querier-deployment.yaml")
+	if err != nil {
+		t.Fatal("Making deployment failed: ", err)
+	}
+
+	if _, err := framework.MonClientV1.ServiceMonitors(ns).Create(s); err != nil {
+		t.Fatal("Creating ServiceMonitor failed: ", err)
+	}
+
+	if err := framework.CreatePrometheusAndWaitUntilReady(ns, basicPrometheus); err != nil {
+		t.Fatal("Creating prometheus failed: ", err)
+	}
+
+	if _, err := testFramework.CreateServiceAndWaitUntilReady(framework.KubeClient, ns, pservice); err != nil {
+		t.Fatal("Creating prometheus service failed: ", err)
+	}
+
+	if err := testFramework.CreateDeployment(framework.KubeClient, ns, thanosQuerier); err != nil {
+		t.Fatal("Creating Thanos querier failed: ", err)
+	}
+
+	if _, err := testFramework.CreateServiceAndWaitUntilReady(framework.KubeClient, ns, tservice); err != nil {
+		t.Fatal("Creating prometheus service failed: ", err)
+	}
+
+	if _, err := testFramework.CreateServiceAndWaitUntilReady(framework.KubeClient, ns, qservice); err != nil {
+		t.Fatal("Creating Thanos querier service failed: ", err)
+	}
+
+	time.Sleep(time.Minute)
```

Comment:
Instead of a sleep I would prefer a `wait.Poll` here to prepare for slow test environments.